source/posix/thread.c

```diff
@@ -0,0 +1,169 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/thread.h>
+
+#include <limits.h>
+#include <errno.h>
+#include <time.h>
+
+struct thread_wrapper {
+    struct aws_allocator *allocator;
+    void(*func)(void *arg);
+    void *arg;
+};
+
+static void *thread_fn(void *arg) {
+    struct thread_wrapper *wrapper = (struct thread_wrapper *)arg;
+    wrapper->func(wrapper->arg);
+    aws_mem_release(wrapper->allocator, wrapper);
+    return NULL;
+}
+
+void aws_thread_clean_up (struct aws_thread *thread) {
+    /* this likely won't do anything if the thread has already terminated, but it shouldn't hurt to send a cancel signal just in case.*/
```

Comment:
Why are we implicitly cancelling the thread? Does the user normally expect that destroying a thread handle would affect the thread state? Who actually writes async-cancel-safe multithreaded code?

Reply:
fair enough, will remove.